File: android/src/main/kotlin/com/gdelataillade/alarm/models/NotificationActionSettings.kt
package com.gdelataillade.alarm.models

import com.google.gson.Gson

data class NotificationActionSettings(
    val hasStopButton: Boolean = false,
    val stopButtonText: String = "Stop"
) {
    companion object {
        fun fromJson(json: Map<String, Any>): NotificationActionSettings {
            val gson = Gson()
            val jsonString = gson.toJson(json)
            return gson.fromJson(jsonString, NotificationActionSettings::class.java)
        }
    }

    fun toJson(): String {
        return Gson().toJson(this)
    }
}

File: ios/Classes/models/NotificationActionSettings.swift
import Foundation

struct NotificationActionSettings: Codable {
    var hasStopButton: Bool = false
    var stopButtonText: String = "Stop"

    static func fromJson(json: [String: Any]) -> NotificationActionSettings {
        return NotificationActionSettings(
            hasStopButton: json["hasStopButton"] as? Bool ?? false,
            stopButtonText: json["stopButtonText"] as? String ?? "Stop"
        )
    }

    static func toJson(notificationActionSettings: NotificationActionSettings) -> [String: Any] {
        return [
            "hasStopButton": notificationActionSettings.hasStopButton,
            "stopButtonText": notificationActionSettings.stopButtonText
        ]
    }
}